internal/provider/resource_ko_resolve.go

```diff
@@ -0,0 +1,263 @@
+package provider
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/google/ko/pkg/build"
+	"github.com/google/ko/pkg/commands"
+	"github.com/google/ko/pkg/commands/options"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"gopkg.in/yaml.v3"
+)
+
+func resolveConfig() *schema.Resource {
+	return &schema.Resource{
+		Description: "",
+
+		CreateContext: resourceKoResolveCreate,
+		ReadContext:   resourceKoResolveRead,
+		DeleteContext: resourceKoBuildDelete,
+
+		Schema: map[string]*schema.Schema{
+			FilenamesKey: {
+				Description: "Filenames, directorys, or URLs to files to use to create the resource",
```

Comment:
I'm pretty sure that ko doesn't support fetching URLs here, even if maybe kubectl does.

Reply:
I copied these descriptions straight from the `ko` cli descriptors 😅 I can change if necessary